src/braket/circuits/circuit.py

```diff
@@ -1235,18 +1309,76 @@ def _generate_frame_wf_declarations(self) -> Optional[str]:
+            # this will change with full parametric calibration support
+            elif (
+                isinstance(instruction.operator, Parameterizable) and gate_calibrations is not None
+            ):
+                self._add_fixed_argument_calibrations(gate_calibrations, instruction)
+        return frames, waveforms
 
-        # Declare the frames and waveforms across all pulse sequences
-        declarable_frames = [f for f in frames.values() if not f.is_predefined]
-        if declarable_frames or waveforms:
-            program = OqpyProgram(None)
-            for f in declarable_frames:
-                program.declare(f._to_oqpy_expression())
-            for wf in waveforms.values():
-                program.declare(wf._to_oqpy_expression())
-            ast = program.to_ast(encal=True, include_externs=False)
-            return ast_to_qasm(ast)
-        return None
+    def _add_fixed_argument_calibrations(
+        self,
+        gate_calibrations: Dict[Tuple[Gate, QubitSet], PulseSequence],
+        instruction: Instruction,
+    ) -> None:
+        """Adds calibrations with arguments set to the instruction parameter values
+
+        Given the collection of parameters in instruction.operator, this function looks for matching
+        parametric calibrations that have free parameters. If such a calibration is found and the
+        number N of its free parameters equals the number of instruction parameters, we can bound
+        the arguments of the calibration and add it to the calibration dictionary.
+
+        If N is smaller, it is probably impossible to assign the instruction parameter values to the
+        corresponding calibration parameters so we raise an error.
+        If N=0, we ignore it as it will not be removed by _generate_frame_wf_defcal_declarations.
+
+        This function modifies its `gate_calibrations` input argument.
+
+        Args:
+            gate_calibrations (Dict[Tuple[Gate, QubitSet], PulseSequence]): a dictionary of
+                calibrations
+            instruction (Instruction): a Circuit instruction
+
+        Raises:
+            NotImplementedError: in two cases: (i) if the instruction contains unbound parameters
+                and the calibration dictionary contains a parametric calibration applicable to this
+                instructions; (ii) if the calibration is defined with a partial number of unbound
+                parameters.
+        """
+        for key, calibration in deepcopy(gate_calibrations).items():
+            gate = key[0]
+            target = key[1]
+            if target != instruction.target:
+                continue
+            if isinstance(gate, type(instruction.operator)) and len(
+                instruction.operator.parameters
+            ) == len(gate.parameters):
+                free_parameter_number = sum(
+                    [not isinstance(p, (float, int, complex)) for p in gate.parameters]
```

Comment:
Should we be checking if it is an instance of FreeParameterExpression instead?

Reply:
Yes, fixed here and the other appearance.